File: 13.调整数值顺序使奇数位于偶数前面/调整数组位置使奇数位于偶数后面.java
public class Solution {
    public void reOrderArray(int [] array) {
        if(array.length == 0){
            return ;
        }
        int left=0;
        int right;
        int len = array.length;
        while(left<len){
            while(left<len&&array[left]%2==1){
                left++;
            }
            right = left+1;
            while(right<len&&array[right]%2==0){
                right++;
            }
            if(right<len){
                int temp = array[right];
                for(int i=right-1; i>=left; i--){
                    array[i+1] = array[i];
                }
                array[left++] = temp;
            }else{
                break;
            }
        }
    }
}

File: 13.调整数值顺序使奇数位于偶数前面/调整数组位置使奇数位于偶数后面.cpp
class Solution {
public:
    void reOrderArray(vector<int> &array) {
        if(array.size()==0){
            return ;
        }
        int left = 0;
        int right;
        while(left < array.size()){
            while(left<array.size()&&array[left]%2==1){
                left++;
            }
            right = left+1;
            while(right < array.size()&&array[right]%2==0){
                right++;
            }
            
            if(right < array.size()){
                int temp = array[right];
                for(int i=right-1; i>=left; i--){
                    array[i+1] = array[i];
                }
                array[left++] = temp;
            }else{
                break;
            }
            
        }
    }
};